### ISSUE TITLE
Support for .publishsettings file for credentials

### ISSUE THREAD

**Issue by adamedx** [chef/chef-provisioning-azure#16]
This driver directly uses the management certificate for credentials. Most people use a .publishsettings file, which contains this certificate.

This file can be obtained via get-azurepublishsettingsfile (part of the azure sdk that you can download on Windows):
https://msdn.microsoft.com/en-us/library/dn495224.aspx

You can install the Azure xplat tools on OS X:
http://go.microsoft.com/fwlink/?linkid=252249&clcid=0x409

Or use npm:
```
sudo npm install -g azure-cli
```

These instructions include what is needed to get a publishsettings file:
http://azure.microsoft.com/en-us/documentation/articles/xplat-cli/#configure

**Comment by stuartpreston** [comment]
I can take a look at this -- as the driver takes a dependency on the Azure SDK for Ruby, the support needs to be added there. I believe there is a current bug with the SDK that prevents certificates in PFX format (such as the one inside the .publishsettings file) being used. Will investigate.

**Comment by adamedx** [comment]
@stuartpreston thank you! A few more tips if you're using a Mac from the doc page listed above:

You can generate a publishsettings file via this command:
`azure account download`

And associate your Azure CLI tools with it via:
`azure account import [path to .publishsettings file]`

**Comment by stuartpreston** [comment]
I put out a new release (0.6.7) of my forked Azure SDK today (https://github.com/stuartpreston/azure-sdk-for-ruby) which has support for specifying a .publishsettings file instead of a .pem or .pfx file.  

The original Azure SDK is now nearly a year behind on its issues and pull requests, hence for expedience we forked it to get it to do the things we need it to. Hopefully one day we can go back to using the official one.

If all goes well then this functionality will work with the next release of the chef-provisioning-azure driver.

**Comment by adamedx** [comment]
Thank you @stuartpreston. In knife-azure, we've been using the REST api directly to stay on top of API changes. I wonder if Fog's Azure support is any closer than the Ruby SDK?

@stuartpreston, do you have PR's against the Azure Ruby SDK?

**Comment by stuartpreston** [comment]
@adamedx fog-azure is taking a dependency on the original Azure SDK so will have a bunch of issues I've now fixed myself in our fork of it (e.g. supporting user captured images, creating user images complete with TCP/UDP endpoints, updated VM role sizes, etc.)

Via Twitter I made contact with Ross Gardler from MSOpenTech who said he was going to try and speak to the maintainers of the Azure Ruby SDK project, but haven't heard anything back so far. There are 21 open PRs and many more open issues against the project currently covering areas I've merged into my own fork. I have not added to this list, just tried to reuse as many of the patches as possible to make it easier to merge later on...  As I see it there's not a lot of point submitting more PRs covering the same changes people have already submitted.

Sounds like you already came to the conclusion to go your own way with knife-azure, it would be good to find a way to share/reuse this work instead of having two different approaches.  Even sharing the test approach would be a win.

Also worth remembering that at some point we'll need to support generating Resource Manager-style requests and this will mean needing to support OAUTH2 (see https://msdn.microsoft.com/en-us/library/azure/dn790557.aspx?f=255&MSPPError=-2147217396) 

Perhaps a continuation of this discussion could be a community session for the open space day at Chefconf?
